### PR TITLE
fix: [CDS-83704]: snow enhacement observation fixes

### PR DESCRIPTION
**Pull request by satyam-harness**
### Summary
Bugs -  There were few bugs found in QE testing of infinite scroll and standard template testings. https://harness.slack.com/archives/C05MVLX4XLG/p1697610225858079

Manual Testing - 

1. Standard templates are displayed in the form template list too and when chosen the value disappears after a few secs, fixed it.
2. Fixed issues with Issue with the search.
3. Search value is cleared on toggling between radio button options.

#### Screenshots

https://github.com/harness/harness-core-ui/assets/96036463/0c7a1c31-1a35-416a-b04d-7caa17f4da29

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
- MessageMetadata: `trigger messagecheck`
- GitLeaks: `trigger gitLeaks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

### CODE REVIEW

**Comment by satyam-harness** [comment]
retrigger jest

**Comment by satyam-harness** [comment]
retrigger cypress-pipeline

**Comment by satyam-harness** [comment]
retrigger eslint

File: src/modules/70-pipeline/components/PipelineSteps/Steps/ServiceNowCreate/ServiceNowCreateStepMode.tsx [hunk 1]
```diff
@@ -526,6 +541,14 @@ function FormContent({
     </Menu>
   )
 
+  const getTemplates = () => {
+    if (fetchingTemplate && !isFetchingTemplateNextTime) {
+      return [{ label: 'Loading Templates...', value: 'Loading Templates...' }]
+    }
+    return defaultTo(templates, [])
+    return templates
```
Comment (chetanvekariya-harness) [inline src/modules/70-pipeline/components/PipelineSteps/Steps/ServiceNowCreate/ServiceNowCreateStepMode.tsx:549]: ??

**Review by chetanvekariya-harness — CHANGES_REQUESTED** [review]

File: src/modules/70-pipeline/components/PipelineSteps/Steps/ServiceNowCreate/ServiceNowCreateStepMode.tsx (hunk 1 above)
Comment (satyam-harness) [reply src/modules/70-pipeline/components/PipelineSteps/Steps/ServiceNowCreate/ServiceNowCreateStepMode.tsx]: solved

**Comment by satyam-harness** [comment]
retrigger eslint

File: src/modules/70-pipeline/components/PipelineSteps/Steps/ServiceNowCreate/ServiceNowCreateStepMode.tsx [hunk 2]
```diff
@@ -840,25 +862,32 @@ function FormContent({
...
                 <div className={cx(stepCss.formGroup, stepCss.lg)}>
                   <FormInput.MultiTypeInput
-                    selectItems={templates}
+                    selectItems={getTemplates()}
                     label={getString('pipeline.serviceNowCreateStep.templateName')}
                     name={`spec.templateName`}
                     key={FieldType.CreateFromTemplate}
-                    placeholder={getString('pipeline.filters.servicePlaceholder')}
+                    placeholder={
+                      selectedTemplateName ? selectedTemplateName : getString('pipeline.filters.servicePlaceholder')
```
Comment (chetanvekariya-harness) [inline src/modules/70-pipeline/components/PipelineSteps/Steps/ServiceNowCreate/ServiceNowCreateStepMode.tsx:870]: Is this intended?

**Review by chetanvekariya-harness — APPROVED** [review]

File: src/modules/70-pipeline/components/PipelineSteps/Steps/ServiceNowCreate/ServiceNowCreateStepMode.tsx (hunk 2 above)
Comment (satyam-harness) [reply src/modules/70-pipeline/components/PipelineSteps/Steps/ServiceNowCreate/ServiceNowCreateStepMode.tsx:870]: Yes

**Review by samarth-sharma31 — APPROVED** [review]

**Comment by satyam-harness** [comment]
trigger messagecheck

**Review by Eric761 — APPROVED** [review]

**Pull request merged**